### PR TITLE
Xraydb

### PR DESCRIPTION
**Pull request by ke-zhang-rd**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by CJ-Wright** [comment]
We (@conda-forge/event-model) accept being maintenaners for this recipe.

**Comment by CJ-Wright** [comment]
There is no license shipped with this there is just a statement in the setup.py saying that it is bsd.

File: recipes/xraydb/setup.py.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+-
+ # copy db from .. to xraydb
+ shutil.copy(os.path.join('..', dbfile), os.path.join('xraydb', dbfile))
+ 
+@@ -35,7 +16,7 @@ setup(name = 'xraydb',
+       author_email = 'newville@cars.uchicago.edu',
+       url          = 'http://github.com/scikit-beam/XrayDB',
+       download_url = 'http://github.com/scikit-beam/XrayDB',
+-      requires = ('numpy', 'sqlalchemy'),
++      requires = ('numpy', 'sqlalchemy', 'sqlalchemy', 'scipy')
```
Comment (mrakitin) [inline recipes/xraydb/setup.py.patch]: Why 2 `sqlalchemy`s are needed?

File: recipes/xraydb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+source:
+  url: https://github.com/scikit-beam/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 0d2899348ffd8914950d4d76f52e5e6446460d25411ab317d2c552da4c8016e0
+  patches:
+    - setup.py.patch
+
+build:
+  number: 0
+  script: cd python && python setup.py install
```
Comment (mrakitin) [inline recipes/xraydb/meta.yaml]: I think the failure of applying the patch in the CI's is due to this "trick", i.e. the pwd is already in the `python/` dir.

File: recipes/xraydb/setup.py.patch [hunk 3]
```diff
@@ -0,0 +1,39 @@
+diff --git a/python/setup.py b/python/setup.py
+index 3824844..1acf773 100644
+--- a/python/setup.py
```
Comment (mrakitin) [inline recipes/xraydb/setup.py.patch]:
```suggestion
--- a/setup.py
```

File: recipes/xraydb/setup.py.patch [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
++++ b/python/setup.py
```
Comment (mrakitin) [inline recipes/xraydb/setup.py.patch]:
```suggestion
+++ b/setup.py
```

File: recipes/xraydb/setup.py.patch (hunk 1 above)
Comment (mrakitin) [inline recipes/xraydb/setup.py.patch]:
```suggestion
+      requires = ('numpy', 'sqlalchemy', 'scipy')
```

**Comment by CJ-Wright** [comment]
I think the issue here is that the tag we're building against doesn't have any python files

**Comment by mrakitin** [comment]
> I think the issue here is that the tag we're building against doesn't have any python files

Oh, right @CJ-Wright. I downloaded the source from https://codeload.github.com/scikit-beam/XrayDB/tar.gz/v1.2, and the archive only has 2 files. Not sure how it happened.

**Comment by mrakitin** [comment]
@ke-zhang-rd, would you be able to follow-up? I think we should now pick https://github.com/xraypy/XrayDB/releases/tag/4.3.1 as a source.

**Pull request closed without merging**